appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb

```diff
@@ -469,24 +431,31 @@
    "source": [
     "%%appyter code_exec\n",
     "# Data transformation for 2D visualization\n",
-    "\n",
     "{% if visualization_method.value == \"All\" or visualization_method.value == \"PCA\" %}\n",
-    "pca = PCA(n_components=min(clustering_topk, len(expr_df)))\n",
-    "transformed_pca = pca.fit_transform(expr_df)\n",
-    "pca_df = create_df(transformed_pca, df_y)\n",
+    "# PCA results have been computed when doing Leiden \n",
+    "pca_df = df_y.copy()\n",
```

Comment:
Can we try and eliminate these copies? Can't you just either use `leiden_df.obsm['']` directly or reasign the variable, i.e. `pca_df = leiden_df.obsm['X_pca']`